src/dotnet/APIView/APIViewWeb/Languages/CppLanguageService.cs

```diff
@@ -0,0 +1,1060 @@
+                            navigateToId = typePart;
+                        }
+                        builder.Append(new CodeFileToken()
+                        {
+                            Kind = CodeFileTokenKind.TypeName,
+                            NavigateToId = navigateToId,
+                            Value = typeValue
+                        });
+                    }
+                    else
+                    {
+                        builder.Text(typePart);
+                    }
+                }
+            }
+        }        
+
+        private static void BuildExpression(CodeFileTokensBuilder builder, CppAstNode exprNode)
+        {
+            switch (exprNode.kind)
+            {
+                case "CStyleCastExpr":
+                case "ConstantExpr":
+                    foreach (var node in exprNode.inner)
+                    {
+                        BuildExpression(builder, node);
+                    }
+                    break;
+                case "ParenExpr":
+                    builder.Punctuation("(");
+                    foreach (var node in exprNode.inner)
+                    {
+                        BuildExpression(builder, node);
+                    }
+                    builder.Punctuation(")");
+                    break;
+                case "IntegerLiteral":
+                    builder.Text(exprNode.value);
+                    break;
+                case "StringLiteral":
+                    builder.Append(exprNode.value, CodeFileTokenKind.StringLiteral);
+                    break;
+                default:
+                    builder.Text(exprNode + " " + exprNode.value);
+                    break;
+            }
+        }
+
+        private class CppAstNode
+        {
+            public string id { get; set; }
+            public string kind { get; set; }
+            public string type { get; set; }
+            public string name { get; set; }
+            public string text { get; set; }
+            public string value { get; set; }
+            public List<CppAstNode> inner { get; set; }
+            public string storageClass { get; set; }
+            public bool? constexpr { get; set; }
+            public bool? inline { get; set; }
+            public bool? isimplicit {get; set; }
+            public bool? isvirtual { get; set; }
+            public bool? ispure { get; set; }
+            public bool? isdefault { get; set; }
+            public bool? isdelete { get; set; }
+            public string tagUsed { get; set; }
+            public string access { get; set; }
+            public string keywords { get; set; }
+            public override string ToString()
+            {
+                return $"{nameof(kind)}: {kind} {nameof(name)}: {name}";
+            }
+        }
+
+        private class CppAstConverter
+        {
+            private static Regex _declKindParser = new Regex("\\w+", RegexOptions.Compiled | RegexOptions.IgnoreCase);
```

Comment:
Why didn't you use a JSON ast?

Reply:
AST-dump has a bug that prevents generating Json for c++.

Reply:
Even in the latest clang? Did we file it. I'm slightly worried that parsing text would be very flaky.

Reply:
Yes. I have used latest clang. We haven't filed it. I will work with Rick or Victor to generate a min repro code and file an issue.